apis/v1alpha2/gateway_types.go

```diff
@@ -144,18 +144,13 @@ type Listener struct {
 	// matched. This field can be omitted for protocols that don't require
 	// hostname based matching.
 	//
-	// Hostname is the fully qualified domain name of a network host, as defined
-	// by RFC 3986. Note the following deviations from the "host" part of the
-	// URI as defined in the RFC:
-	//
-	// 1. IP literals are not allowed.
-	// 2. The `:` delimiter is not respected because ports are not allowed.
-	//
-	// Hostname can be "precise" which is a domain name without the terminating
-	// dot of a network host (e.g. "foo.example.com") or "wildcard", which is a
-	// domain name prefixed with a single wildcard label (e.g. `*.example.com`).
-	// The wildcard character `*` must appear by itself as the first DNS label
-	// and matches only a single label.
+	// For HTTPRoute objeccts, there is an interaction with the
+	// `spec.hostnames` array. In short, if a hostname is specified in the
+	// Listener, then all the HTTPRoute's hostnames must match the hostname
+	// specified here, including both precise and wildcard hostname types.
+	// See the HTTPRoute `spec.hostname` fields for information about the exact
+	// rules used, and see the `Hostname` type for definitions of `precise` and
+	// `wildcard` hostnames.
```

Comment:
I moved the rules for hostnames to the `Hostname` struct, but wanted to note the possible interaction here, as I felt that means we are covered for both users (cluster admin and application developer). I can see an argument for not having this here though.